### ISSUE TITLE
Correctly Naming Charts and Dashboard Organization

### ISSUE THREAD

**Issue by birbjam** [BNHM/AmphibiaWebDiseasePortal#41]
- [x] Organize charts into drop downs that make the most sense for the type of data displayed
- [x] Rename charts so everything is named correctly according to what it represents
- [x] Update bar/pie chart colors to make sure they are consistent
-- [ ]

**Comment by mkoo** [comment]
A couple of comments/ thoughts on the Charts:
 - we should have consistent colors for the data but distinguishing when positive/ negative disease data is stacked. If we show both Bd/Bsal stacked together then should be two different colors than what is used in Positive/ Negative. Try something from ColorBrewer? (green/orange or purple?) https://colorbrewer2.org/#type=qualitative&scheme=Dark2&n=3
- rename True/False to Positive/ Negative
 - We need a button to clear current state (otherwise acts unpredictably if several are chosen)
- Merge Bd Counts by country and Bd Detected by Country (likewise for Bsal and for Both)--> this was an issue in other portal. We only need to see Bd/Bsal/Both and whether Pos/Neg per country, since plain vanilla counts is not intrinsically interesting.
 - can histogram y-axis be ordered (left to right) in descending values?
- Pie Charts should be in a separate drop-down as it's confusing to have histograms mixed in piecharts. I also dont know the value of showing all the portal data in a giant piechart. Can we show piecharts per country or per species yet? (eg. https://amphibiandisease.org/dashboard.php?taxon=Acris+crepitans) That would be more value!
- Or maybe better: clicking on a bar in Bd Detected by Country histogram generates a piechart for that country would be great!
- Remove Bd/Bsal/Both by year and leave only Bd/Bsal/Both detected by year (same issue as Counts by country)
- Why does Bd and Bsal detected by year look identical? (Bsal should start in 2013 earliest)
- I love the idea of showing data by Project but I dont think by the ID is going to be popular. Maybe remove from the general data dashboard and display on the Project detail page instead?

excited to see this progress!

**Comment by birbjam** [comment]
Great, this is exactly the feedback I was hoping for! :)

My initial plan was to just create charts using all the data provided first and then kind of organize everything into what makes sense.

- The Bd/Bsal by year charts are identical because the data in the API is duplicated. I sent John an email about it so that should be an easy fix cause once the correct data is uploaded the chart should automatically generate with the correct values.
- I moved the data charts for projects to the individual project pages, so each project detail page that has data shows a chart. I originally wanted to do that anyway so that's perfect. 

I'll work on the rest of these issues this week as well.

**Comment by mkoo** [comment]
Glad you like feedback! Since this issue includes "Dashboard Org" I'll include here too.

As with the current site, high-level portal stats and a top ten list and links would be milestones for the first iteration of this version. Honestly just copying pasting into the new Charts page would be fine.

Specifically (available in the API "scientificName_Both.json" with some number munging):
![Firefox_Screenshot_2020-05-06T23-20-51 268Z](https://user-images.githubusercontent.com/2523089/81237896-93d0ab00-8fb5-11ea-95c1-240042ae5867.png)

**Comment by birbjam** [comment]
I also noticed that the old dashboard included a list of species sampled and each species being linked to its corresponding AmphibiaWeb page? Do we want to do something like that as well?

**Comment by mkoo** [comment]
That would be great to migrate over-- doable?

As I've mentioned before we are able to create workable permalinks to
species data like this:
https://amphibiandisease.org/dashboard.php?taxon=Afrixalus+lacteus
which were embedded in AmphibiaWeb. I've removed them since those urls will
not make sense but if we can recreate something else predictable based on
genus+speciesepithet that would be ideal.

Basically the idea for the first iteration is to recreate the functions and
features of the current portal as much as possible!

**Comment by birbjam** [comment]
Okay yeah I think adding links to AmphibiaWeb should be doable. I'll look into it.

I think I fixed the issue of the charts acting wonky when multiple ones are selected so now you can only select one at a time and the charts won't change when you hover over them.

**Comment by mkoo** [comment]
Some interface notes:
- I preferred the previous way you had the table of counts visible at the top immediately. are these counts not cached?
- Remove the Chart tab since it doesnt do anything
- nice start to the species list-- maybe call the tab ' Species' or 'List by Species'?
- Sizing and columns are off on the table for Top Ten Countries and Top Ten Species (should also have that labeled)
![Firefox_Screenshot_2020-05-19T18-20-59 327Z](https://user-images.githubusercontent.com/2523089/82363681-016fd480-99c3-11ea-8cd2-977aca9e07ce.png)
- Charts: I think each graph that shows Bd or Bsal should be stacked positive and negative (red/blue); each chart that shows 'Both' pathogen should show Bd and Bsal data as stacked so they are differentiated (orange/purple). thus you would remove the 'Both' unstacked options.
- Also, I asked earlier: can the y-axis be ordered (left to right) in descending values? (is this possible?)

thanks!

**Comment by birbjam** [comment]
- I'm not sure what you mean about the counts being cached or that the Chart tab doesn't do anything? 

- The idea was to separate tables from charts and have a list tab of species so it all appeared cleaner but I can put the tables back with the charts if that is preferred. I was thinking that it would be easier to separate them than scroll past a bunch of tables to get to the charts? 

- Not sure why the tables look the way you see them, I viewed them in mobile and on different sized screens and they don't look like that. I'll check them with different browsers and see if that changes what they look like.

- I'll rename the List Tab, label the tables, organize the charts further, and y-axis ordering is also on my to-do list. 

Thanks for the feedback!

**Comment by birbjam** [comment]
Ah. The tables appearing the way that they are is a Firefox issue. I'll figure out how to fix that.

**Comment by mkoo** [comment]
ok, I see the Chart tab issue was a state issue-- I was in Tables, then Charts, then back again when the Chart tab wasnt doing anything since it was open already. Reloaded page to restore!

**Comment by birbjam** [comment]
@mkoo I'm working on sorting the charts by order descending, should this also include data displayed by year?

**Comment by mkoo** [comment]
good call-- not by year; keep chronological!
